rules.d/60-persistent-storage.rules

```diff
@@ -93,6 +93,9 @@ ENV{DEVTYPE}=="disk", DEVPATH!="*/virtual/*", IMPORT{builtin}="path_id"
 KERNEL=="mmcblk[0-9]boot[0-9]", ENV{DEVTYPE}=="disk", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}-boot%n"
 KERNEL!="mmcblk[0-9]boot[0-9]", ENV{DEVTYPE}=="disk", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}"
 ENV{DEVTYPE}=="partition", ENV{ID_PATH}=="?*", SYMLINK+="disk/by-path/$env{ID_PATH}-part%n"
+# ATA device compatible links
+KERNEL!="mmcblk[0-9]boot[0-9]", ENV{DEVTYPE}=="disk", ENV{ID_PATH}=="*ata*", PROGRAM="/bin/sh -c 'compath=$env{ID_PATH};echo ${compath%\.[0-9]*'",ENV{ID_PATH_ATA_COMPAT}="%c", SYMLINK+="disk/by-path/$env{ID_PATH_ATA_COMPAT}"
+ENV{DEVTYPE}=="partition", ENV{ID_PATH}=="*ata*", PROGRAM="/bin/sh -c 'compath=$env{ID_PATH};echo ${compath%\.[0-9]*'", ENV{ID_PATH_ATA_COMPAT}="%c", SYMLINK+="disk/by-path/${$env{ID_PATH_ATA_COMPAT}-part%n"
```

Comment:
There seems to be a missing closing brace for the opening brace in `${compath`, in both lines. So both of those commands will fail. Was this tested at all?

Reply:
I am really sorry for the mistake.  I did some tests before sending it for review, but 
apparently I failed to copy the commands in the rules file from my test machine to 
my patch correctly,  because I used a different way to get the $compath than in the 
beginning.  Here is the test I did:
Currently the link under /dev/disk/by-path for ATA device is 
/dev/disk/by-path/pci-0000:00:0b.0-ata-6  or
/dev/disk/by-path/pci-0000:00:0b.0-ata-6-part1
With this patch, it will be
/dev/disk/by-path/pci-0000:00:0b.0-ata-6
/dev/disk/by-path/pci-0000:00:0b.0-ata-6.0 or
/dev/disk/by-path/pci-0000:00:0b.0-ata-6-part1
/dev/disk/by-path/pci-0000:00:0b.0-ata-6.0-part1

The following part is the debug messages when udev rules file processes the ATA device

IMPORT builtin 'path_id' /usr/lib/udev/rules.d/60-persistent-storage.rules:68
LINK 'disk/by-path/pci-0000:00:0b.0-ata-6.0' /usr/lib/udev/rules.d/60-persistent-storage.rules:69
PROGRAM '/bin/sh -c 'compath=pci-0000:00:0b.0-ata-6.0;echo ${compath%\.[0-9]*}' /usr/lib/udev/rules.d/60-persistent-storage.rules:73
starting '/bin/sh -c 'compath=pci-0000:00:0b.0-ata-6.0;echo ${compath%\.[0-9]*}'
'/bin/sh -c 'compath=pci-0000:00:0b.0-ata-6.0;echo ${compath%\.[0-9]*}'(out) 'pci-0000:00:0b.0-ata-6'
Process '/bin/sh -c 'compath=pci-0000:00:0b.0-ata-6.0;echo ${compath%\.[0-9]*}' succeeded.
LINK 'disk/by-path/pci-0000:00:0b.0-ata-6' /usr/lib/udev/rules.d/60-persistent-storage.rules:73